src/graph/RadarPostprocessPointsNode.cpp

```diff
@@ -0,0 +1,195 @@
+		fieldData->resize(filteredIndices->getCount(), false, false);
+		cacheManager.insert(field, fieldData, true);
+	}
+
+	if (!cacheManager.isLatest(field)) {
+		auto fieldData = cacheManager.getValue(field);
+		fieldData->resize(filteredIndices->getCount(), false, false);
+		char* outPtr = static_cast<char*>(fieldData->getRawWritePtr());
+		auto fieldArray = input->getFieldData(field);
+		if (!isDeviceAccessible(fieldArray->getMemoryKind())) {
+			auto msg = fmt::format("RadarPostprocessPoints requires its input to be device-accessible, {} is not", field);
+			throw InvalidPipeline(msg);
+		}
+		const char* inputPtr = static_cast<const char*>(fieldArray->getRawReadPtr());
+		gpuFilter(getStreamHandle(), filteredIndices->getCount(), filteredIndices->getReadPtr(), outPtr, inputPtr,
+		          getFieldSize(field));
+		CHECK_CUDA(cudaStreamSynchronize(getStreamHandle()));
+		cacheManager.setUpdated(field);
+	}
+
+	return cacheManager.getValue(field);
+}
+
+std::vector<rgl_field_t> RadarPostprocessPointsNode::getRequiredFieldList() const { return {DISTANCE_F32, AZIMUTH_F32}; }
+
+// RadarCluster methods implementation
+
+RadarPostprocessPointsNode::RadarCluster::RadarCluster(Field<RAY_IDX_U32>::type index, float distance, float azimuth)
+{
+	indices.emplace_back(index);
+	minMaxDistance = {distance, distance};
+	minMaxAzimuth = {azimuth, azimuth};
+}
+
+void RadarPostprocessPointsNode::RadarCluster::addPoint(Field<RAY_IDX_U32>::type index, float distance, float azimuth)
+{
+	indices.emplace_back(index);
+	minMaxDistance[0] = std::min(minMaxDistance[0], distance);
+	minMaxDistance[1] = std::max(minMaxDistance[1], distance);
+	minMaxAzimuth[0] = std::min(minMaxAzimuth[0], azimuth);
+	minMaxAzimuth[1] = std::max(minMaxAzimuth[1], azimuth);
+}
+
+inline bool RadarPostprocessPointsNode::RadarCluster::isCandidate(float distance, float azimuth, float distanceSeparation,
+                                                                  float azimuthSeparation) const
+{
+	return (distance >= minMaxDistance[0] - distanceSeparation && distance <= minMaxDistance[1] + distanceSeparation) &&
+	       (azimuth >= minMaxAzimuth[0] - azimuthSeparation && azimuth <= minMaxAzimuth[1] + azimuthSeparation);
+}
+
+inline bool RadarPostprocessPointsNode::RadarCluster::canMergeWith(const RadarCluster& other, float distanceSeparation,
+                                                                   float azimuthSeparation) const
+{
+	bool isDistanceGood = std::abs(minMaxDistance[0] - other.minMaxDistance[1]) <= distanceSeparation &&
+	                      std::abs(minMaxDistance[1] - other.minMaxDistance[0]) <= distanceSeparation;
+
+	bool isAzimuthGood = std::abs(minMaxAzimuth[0] - other.minMaxAzimuth[1]) <= azimuthSeparation &&
+	                     std::abs(minMaxAzimuth[1] - other.minMaxAzimuth[0]) <= azimuthSeparation;
+
+	return isDistanceGood && isAzimuthGood;
+}
+
+void RadarPostprocessPointsNode::RadarCluster::mergeWith(RadarCluster other)
```

Comment:
I would consider passing other as const& - you move indices below but you copy it here either way. Then you should go with e.g. std::transform instead std::move in line 194.

I would also consider mergeIndicesFrom name instead of mergeWith.